src/azul/deployment.py

```diff
@@ -434,8 +436,46 @@ def client(self, *args, **kwargs):
         Caching the result of this function is not necessary and will be harmful
         if the cached value is used by a thread other than the one that called
         this function.
+
+        :param azul_logging: whether to log the client's requests and responses
+        """
+        client = self.boto3_session.client(*args, **kwargs)
+        if azul_logging:
+            client.meta.events.register('before-send', self._log_client_request)
+            client.meta.events.register('response-received', self._log_client_response)
+        return client
+
+    def _log_client_request(self,
+                            event_name: str,
+                            request: Any,
```

Comment:
Is there a more specific type that would be appropriate here? We could investigate by temporarily adding `type(Request)` to the log.

Reply:
It seems like `AWSPreparedRequest` is the type we need.